fe/fe-core/src/main/java/com/starrocks/sql/optimizer/property/ValuePropertyDeriver.java

```diff
@@ -0,0 +1,194 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.property;
+
+import com.google.common.collect.Maps;
+import com.starrocks.catalog.FunctionSet;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CallOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CompoundPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CompoundPredicateOperator.CompoundType;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.InPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.IsNullPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperatorVisitor;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ValuePropertyDeriver extends ScalarOperatorVisitor<Map<ScalarOperator, ScalarOperator>, Void> {
+
+    public ValueProperty derive(ScalarOperator scalarOperator) {
+        Map<ScalarOperator, ScalarOperator> valueMap = scalarOperator.accept(this, null);
+        Map<ScalarOperator, ValueProperty.ValueWrapper> newMap = valueMap.entrySet().stream()
+                .filter(e -> !e.getValue().equals(ConstantOperator.TRUE))
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> new ValueProperty.ValueWrapper(entry.getValue())));
+        return new ValueProperty(newMap);
+    }
+
+
+    @Override
+    public Map<ScalarOperator, ScalarOperator> visit(ScalarOperator scalarOperator, Void context) {
+        return Maps.newHashMap();
+    }
+
+    @Override
+    public Map<ScalarOperator, ScalarOperator> visitBinaryPredicate(BinaryPredicateOperator binaryPredicate, Void context) {
```

Comment:
the implement like RangeExtractor?

Reply:
A little difference.  We primarily use this to generate new predicates to describe this column. For example, `(colA > 1 and coB = 1) or (colA = -1 and colB = 2)` will generate two predicates, `colA > 1 or colA = -1` and `colB = 1 or colB = 2`, respectively, to describe the possible values of colA and colB. Then, we yeild range information from the new predicates.

Reply:
emm, I think RangeExtractor could extract `colA >= -1` and `colB in (1, 2)`  from `(colA > 1 and coB = 1) or (colA = -1 and colB = 2)` directly ?

Reply:
```
 private static class RangeExtractor extends ScalarOperatorVisitor<Void, Void> {
        private final Map<ScalarOperator, ValueDescriptor> descMap = Maps.newHashMap();
```
The RangeExtractor just extract the valueDescriptor.  We need obtain the new predicate like `colA > 1 or colA = -1` which will help dervie predicate based on join on predicate like `colA = colB`.